### ISSUE TITLE
Errors Encountered on Humble

### ISSUE THREAD

**Issue by peterdavidfagan** [YuLiHN/kinect2_ros2#3]
Hi @YuLiHN, 

Thank you so much for adding support for ROS2. 

**Build Error:**
I am currently facing an issue in the build phase, I have the same system specs as those you outlined in the README.md file.

```
Starting >>> kinect2_registration
--- stderr: kinect2_registration                         
CMake Error at CMakeLists.txt:104 (message):
  message called with incorrect number of arguments


---
Failed   <<< kinect2_registration [1.02s, exited with code 1]

Summary: 0 packages finished [1.15s]
  1 package failed: kinect2_registration
  1 package had stderr output: kinect2_registration
  1 package not processed
```

I was able to resolve this through removing https://github.com/YuLiHN/kinect2_ros2/blob/252ff5a741ebdca334487be4e0681ed014d6cb2d/kinect2_registration/CMakeLists.txt#L104.

**Runtime Error:**

```
[kinect2_bridge-1] /home/peter/Code/ros/kinect_ws/install/kinect2_bridge/lib/kinect2_bridge/kinect2_bridge: error while loading shared libraries: libfreenect2.so.0.2: cannot open shared object file: No such file or directory
```

Resolved by creating a symbolic link between shared library location and usr lib folder. 

```
sudo ln -s $HOME/freenect2/lib/libfreenect2.so.0.2 /usr/lib/libfreenect2.so.0.2
```

Within the launch file I also had to use the default serial number for the file to work.

**Comment by YuLiHN** [comment]
> Hi @YuLiHN,
> 
> Thank you so much for adding support for ROS2.
> 
> **Build Error:** I am currently facing an issue in the build phase, I have the same system specs as those you outlined in the README.md file.
> 
> ```
> Starting >>> kinect2_registration
> --- stderr: kinect2_registration                         
> CMake Error at CMakeLists.txt:104 (message):
>   message called with incorrect number of arguments
> 
> 
> ---
> Failed   <<< kinect2_registration [1.02s, exited with code 1]
> 
> Summary: 0 packages finished [1.15s]
>   1 package failed: kinect2_registration
>   1 package had stderr output: kinect2_registration
>   1 package not processed
> ```
> 
> I was able to resolve this through removing
> 
> https://github.com/YuLiHN/kinect2_ros2/blob/252ff5a741ebdca334487be4e0681ed014d6cb2d/kinect2_registration/CMakeLists.txt#L104
> 
> .
> **Runtime Error:**
> 
> ```
> [kinect2_bridge-1] /home/peter/Code/ros/kinect_ws/install/kinect2_bridge/lib/kinect2_bridge/kinect2_bridge: error while loading shared libraries: libfreenect2.so.0.2: cannot open shared object file: No such file or directory
> ```
> 
> Resolved by creating a symbolic link between shared library location and usr lib folder.
> 
> ```
> sudo ln -s $HOME/freenect2/lib/libfreenect2.so.0.2 /usr/lib/libfreenect2.so.0.2
> ```
> 
> Within the launch file I also had to use the default serial number for the file to work.

Hey Peter,
I tried to reproduce your building problem but I can still build even on a fresh new computer without removing this line
https://github.com/YuLiHN/kinect2_ros2/blob/252ff5a741ebdca334487be4e0681ed014d6cb2d/kinect2_registration/CMakeLists.txt#L104

But thanks for pointing that out maybe it will help someone use this repo later on.

For the link problem I will add it to readme.

And sorry I have not ported kinect2_calibration packages to ros2 because I was so busy in the past months. Any pull request would be welcome. But you can calibrate your camera using ros1 and put the file in kinect2_bridge/data/ then change to your serial number here https://github.com/YuLiHN/kinect2_ros2/blob/252ff5a741ebdca334487be4e0681ed014d6cb2d/kinect2_bridge/launch/kinect2_bridge.launch.py#L11

Hope you enjoy!

**Comment by leahcornelius** [comment]
Hi, yep i have the same issue. Symblink has already been created but get "message called with incorrect number of arguments" when running colcon build (ros humble). Ill try removing the link and see the results.

**Comment by leahcornelius** [comment]
Ah ok so found the problem. If the variable MODULE_LIBS is empty, then no arguments are passed to the message function and hence the error above. Rather than removing the message line you can just put " marks around the ${...} bit which means a blank line will be printed if it is unset/empty rather than an error. Ill open a PR adding this, but ill need @YuLiHN to confirm the function of this variable, and if it being unset is considered normal behaviour or if it is a cause for error (in which case I will make it print a clearer error  fatal message rather than make raising a error)

**Comment by leahcornelius** [comment]
I also have the issue with the launch file regarding the serial number. The default one doesn't work as it says device with serial X not found. If I change it to all 0s the same error. Do I need to use protonect to get the serial number and use that?

**Comment by YuLiHN** [comment]
> I also have the issue with the launch file regarding the serial number. The default one doesn't work as it says device with serial X not found. If I change it to all 0s the same error. Do I need to use protonect to get the serial number and use that?

try to comment this line
https://github.com/YuLiHN/kinect2_ros2/blob/c3fa8649892dcaf7c448378c49fcbc5dfbc4be8c/kinect2_bridge/launch/kinect2_bridge.launch.py#L11

or change it to 'sensor' : ''

to see if it works.

**Comment by leahcornelius** [comment]
Yep, that makes it work just fine. Im now trying to get performance to half decent levels but ill open a separate issue for that

**Comment by JJTovar15** [comment]
Hi there I'm having trouble executing it in RVIZ it says the following:

New publisher discovered on topic '/kinect2/hd/points', offering incompatible QoS. No messages will be sent to it. Last incompatible policy: RELIABILITY_QOS_POLICY

Is there some workaorund to this?. Thanks
